### PR TITLE
Feature/authentication

### PR DESCRIPTION
**Pull request by wansom**
new feature-authentication  added

### CODE REVIEW

**Comment by CarltonK** [comment]
@wansom Have you already tested this locally?
Otherwise if you want to test using flutter locally
Try this article
https://fireship.io/snippets/firestore-emulator-flutter/

File: functions/src/handlers/firestore_user.ts [hunk 1]
```diff
@@ -4,18 +4,91 @@ import { firestore } from 'firebase-admin';
...
         this.logger.setLogLevel('debug');
     }
 
     public async newUserDocumentHandler(snapshot: firestore.QueryDocumentSnapshot, context: functions.EventContext) {
+        if (snapshot.data().role == 'nonpractitioner') {
+            try {
+                this.logger.info('The new user is identified by: ', snapshot.data().uid);
+                //set public profile
+                this.db.doc(`users/${snapshot.data().uid}/public_profile/${snapshot.data().uid}`).set({
+                    Name: snapshot.data().name,
```
Comment (CarltonK) [inline functions/src/handlers/firestore_user.ts:19]: @wansom Could you also rename these variables to the corresponding variable names on the flutter models?

**Review by CarltonK — CHANGES_REQUESTED** [review]

File: functions/src/handlers/firestore_user.ts (hunk 1 above)
Comment (wansom) [reply functions/src/handlers/firestore_user.ts]: i have,could you check the latest push?

**Comment by wansom** [comment]
I have made the changes if the logic implementation is okay, I could start
on the frontend implementation

**Comment by CarltonK** [comment]
@wansom Yeah that's alright. Make sure to use the correct branch naming

**Pull request merged**